replica/database.hh

```diff
@@ -326,6 +326,40 @@ struct cf_stats {
     uint64_t total_view_updates_on_wrong_node = 0;
 };
 
+/*!
+ * \brief capacity_unit_holder represents a mechanism for tracking the consumption of capacity units.
+ *
+ * Capacity units are simplified representations of user operations within the system.
+ * Each operation performed by a user consumes a specific number of
+ * capacity units, depending on its complexity and resource requirements.
+ *
+ * The `capacity_unit_holder` class encapsulates the concept of capacity units.
+ *
+ * This class is **protocol-agnostic**, meaning it does not depend on any specific communication protocol.
+ * Each protocol that utilizes this class will define how the capacity units are updated and managed within
+ * its context.
+ *
+ * Capacity units serve two primary purposes:
+ *
+ * 1. Billing: They provide a standardized way to quantify resource usage, enabling accurate and transparent
+ *    billing based on the capacity units consumed by a user's operations.
+ *
+ * 2. Usage Limitations: They help enforce usage policies by limiting the number of capacity units
+ *    a user can consume within a given time frame.
+ *
+ */
+class capacity_unit_holder {
+    uint64_t total_units = 0;
+public:
+    uint64_t get_units() const noexcept {
+        return total_units;
+    }
+    uint64_t add_units(uint64_t units) noexcept {
+        total_units += units;
+        return total_units;
+    }
+};
```

Comment:
I wouldn't call it a "holder", it's a counter. I think we use the word "holder" is other contexts to keep some sort of RAII object alive.
By the way, do we need a type for this specific counter - as far as I know all other counters used in metrics are just uint64_t and code does "count += 7" without calling a function.

Reply:
The reason for introducing the type is that, in the future, I'll be able to add more logic to it. Specifically, the cap functionality can be part of this object

Reply:
So maybe capacity_unit_processor will be a better name? Something that will suggest that it can be used also for something else than just holding the counter value.